src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -0,0 +1,214 @@
+
+    return esClient.performRequest(bulkRequest);
+  }
+
+  public UserTerm toUserTerm(User user) {
+    if (Objects.isNull(user)) {
+      return null;
+    }
+
+    UserTerm term = new UserTerm();
+
+    term.setUserId(user.getUserId());
+
+    return term;
+  }
+
+  public StudyTerm toStudyTerm(Study study) {
+    if (Objects.isNull(study)) {
+      return null;
+    }
+
+    StudyTerm term = new StudyTerm();
+
+    term.setDescription(study.getDescription());
+    term.setStudyName(study.getName());
+    term.setStudyId(study.getStudyId());
+    term.setDataTypes(study.getDataTypes());
+    term.setPiName(study.getPiName());
+    term.setPublicVisibility(study.getPublicVisibility());
+
+    findStudyProperty(
+        study.getProperties(), "phenotypeIndication"
+    ).ifPresent(
+        prop -> term.setPhenotype((String) prop.getValue())
+    );
+
+    findStudyProperty(
+        study.getProperties(), "species"
+    ).ifPresent(
+        prop -> term.setSpecies((String) prop.getValue())
+    );
+
+    findStudyProperty(
+        study.getProperties(), "dataCustodianEmail"
+    ).ifPresent(
+        prop -> term.setDataCustodian((String) prop.getValue())
+    );
+
+    if (Objects.nonNull(study.getCreateUserId())) {
+      User dataSubmitter = userDAO.findUserById(study.getCreateUserId());
+      term.setDataSubmitter(toUserTerm(dataSubmitter));
+    }
+
+    return term;
+  }
+
+  public DatasetTerm toDatasetTerm(Dataset dataset) {
+    if (Objects.isNull(dataset)) {
+      return null;
+    }
+
+    DatasetTerm term = new DatasetTerm();
+
+    term.setDatasetId(dataset.getDataSetId());
+    term.setDatasetIdentifier(dataset.getDatasetIdentifier());
+
+    if (Objects.nonNull(dataset.getStudy())) {
+      term.setStudy(toStudyTerm(dataset.getStudy()));
+    }
+
+    term.setDacId(dataset.getDacId());
+
+    Collection<Integer> approvedUserIds =
+        dataAccessRequestDAO.findAllUserIdsWithApprovedDARsByDatasetId(
+            dataset.getDataSetId());
+
+    if (!approvedUserIds.isEmpty()) {
+      Collection<User> approvedUsers = userDAO.findUsers(approvedUserIds);
+
+      List<UserTerm> approvedUserTerms = approvedUsers.stream().map(this::toUserTerm).toList();
+      term.setApprovedUsers(approvedUserTerms);
+    }
+
+    if (Objects.nonNull(dataset.getDataUse())) {
+      term.setDataUse(useRestrictionConverter.translateDataUseSummary(dataset.getDataUse()));
+    }
+
+    findDatasetProperty(
+        dataset.getProperties(), "openAccess"
+    ).ifPresent(
+        datasetProperty -> term.setOpenAccess((Boolean) datasetProperty.getPropertyValue())
+    );
+
+    findDatasetProperty(
+        dataset.getProperties(), "numberOfParticipants"
+    ).ifPresent(
+        datasetProperty -> term.setParticipantCount((Integer) datasetProperty.getPropertyValue())
+    );
+
+    findDatasetProperty(
+        dataset.getProperties(), "url"
+    ).ifPresent(
+        datasetProperty -> term.setUrl((String) datasetProperty.getPropertyValue())
```

Comment:
Here and in other places, use `getPropertyValueAsString()` as opposed to casting. We should try to avoid the casts here and in other places as well.